game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleExecutablesTest.java

```diff
@@ -0,0 +1,1670 @@
+    step.execute(null, delegateBridge);
+
+    if (attacker) {
+      verify(battle).firstStrikeAttackersFire(returnFire);
+    } else {
+      verify(battle).firstStrikeDefendersFire(returnFire);
+    }
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then attacker has return fire all")
+  void firstStrikeAttackerReturnFireAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, true, true, true, true),
+        MustFightBattle.ReturnFire.ALL,
+        true);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then defender has return fire all")
+  void firstStrikeDefenderReturnFireAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, true, true, true, true),
+        MustFightBattle.ReturnFire.ALL,
+        false);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is true, then attacker>standard>defender")
+  void firstStrikeOrderAttHasDestroyerDefHasDestroyerWW2v2FalseSneakAttackTrue() {
+    assertThatFirstStrikeStepOrder(
+        givenFirstStrikeBattleSetup(true, true, false, true, false),
+        List.of(
+            FirstStrikeBattleStep.ATTACKER,
+            FirstStrikeBattleStep.STANDARD,
+            FirstStrikeBattleStep.DEFENDER));
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is true, then attacker has return fire all")
+  void firstStrikeAttackerReturnFireAttHasDestroyerDefHasDestroyerWW2v2FalseSneakAttackTrue() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, true, false, true, false),
+        MustFightBattle.ReturnFire.ALL,
+        true);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is true, then defender has return fire all")
+  void firstStrikeDefenderReturnFireAttHasDestroyerDefHasDestroyerWW2v2FalseSneakAttackTrue() {
+    assertThatFirstStrikeReturnFireIs(
+        givenFirstStrikeBattleSetup(true, true, false, true, false),
+        MustFightBattle.ReturnFire.ALL,
+        false);
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is false, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is false, then attacker>standard>defender")
+  void firstStrikeOrderAttHasDestroyerDefHasDestroyerWW2v2FalseSneakAttackFalse() {
+    assertThatFirstStrikeStepOrder(
```

Comment:
CONSIDER THIS COMMENT FIRST!

After reading through most of the tests, the pattern of testing I think has become more clear. Have you considered parameterizing the whole thing?

If you had a value object for the battle setup, then it should become pretty easy!

Example:
```
@ParameterizedTest
@MethodSource
void verifyFirstStrikeBattleScenarios( 
     BattleSetupParameters setup, MustFightBattle returnFire, boolean attackersExpectedToHaveFirstStrike) {

   verifyFirstStrikebattle(
        setup,  returnFire,  attackersExpectedToHaveFirstStrike);
}


static List<Arguments> verifyFirstStrikeBattleScenarios() {
  return List.of(
     Arguments.of(BattleSetupParameters.builder()....build(), MustFightBattle.ReturnFire.ALL, false),
     :
     :
     );
}
```

Reply:
Yes, I had thought of parameterizing it.  I'll do that next time.